<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClassInstanceDialog</class>
 <widget class="QDialog" name="ClassInstanceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Type:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_instanceType">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Name:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLineEdit" name="lineEdit_name"/>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>ID:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_id">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Data Members:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="m3d::TableWidgetMod" name="tableWidget">
     <property name="editTriggers">
      <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
     </property>
     <property name="tabKeyNavigation">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <column>
      <property name="text">
       <string>Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Type</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_hideMembers">
     <property name="text">
      <string>Hide _prefixed Data Members</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="pushButton_openOwner">
       <property name="toolTip">
        <string>Open owning chip's dialog box</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="editorapp.qrc">
         <normaloff>:/EditorApp/Resources/dialog.png</normaloff>:/EditorApp/Resources/dialog.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_copyRef">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Copy Reference</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="editorapp.qrc">
         <normaloff>:/EditorApp/Resources/copy.png</normaloff>:/EditorApp/Resources/copy.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_destroy">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Destroy Instance</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="editorapp.qrc">
         <normaloff>:/EditorApp/Resources/delete.png</normaloff>:/EditorApp/Resources/delete.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="toolTip">
        <string>Accept changes and close dialog box.</string>
       </property>
       <property name="whatsThis">
        <string>Applies the settings and closes the dialog box.</string>
       </property>
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>m3d::TableWidgetMod</class>
   <extends>QTableWidget</extends>
   <header>tablewidgetmod.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="editorapp.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>tableWidget</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>onTableDoubleClick(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_hideMembers</sender>
   <signal>clicked()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>hideMembersClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ClassInstanceDialog</sender>
   <signal>finished(int)</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>onFinish(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>185</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>okButton</sender>
   <signal>clicked()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>396</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_openOwner</sender>
   <signal>clicked()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>openOwner()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_copyRef</sender>
   <signal>clicked()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>copyRef()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_destroy</sender>
   <signal>clicked()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>destroyRef()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_name</sender>
   <signal>editingFinished()</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>nameChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>197</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>ClassInstanceDialog</receiver>
   <slot>onCellChanged(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onTableDoubleClick(int,int)</slot>
  <slot>hideMembersClicked()</slot>
  <slot>onFinish(int)</slot>
  <slot>openOwner()</slot>
  <slot>nameChanged()</slot>
  <slot>copyRef()</slot>
  <slot>destroyRef()</slot>
  <slot>onCellChanged(int,int)</slot>
 </slots>
</ui>
